### PR TITLE
Added user authentication method permissions

### PR DESCRIPTION
**Pull request by mmcla**
Added new userauthenticationmethod.* permission scopes.

### CODE REVIEW

File: concepts/permissions-reference.md [hunk 1]
```diff
@@ -1639,6 +1639,34 @@ The *CreatedByApp* constraint associated with this permission indicates the serv
 
 ---
 
+## User authentication method permissions (preview)
+
+#### Delegated permissions
+
+|Permission                              |Display String                        |Description        |Admin Consent Required | Microsoft Account supported |
+|:---------------------------------------|:-------------------------------------|:------------------|:----------------------|:----------------------------|
+|_UserAuthenticationMethod.ReadWrite.All_|Manage users' authentication methods  |Allows create/read/update/delete for authentication methods and default method on a user. Users authorized to do so can perform this for many or all users; users without those roles can only perform the action on self. This permission is currently in private preview only and is not yet available for public use.|Yes|No|
```
Comment (dkershaw10) [inline concepts/permissions-reference.md]: Please re-order to have the least privileged first, and group (like other sections):

UserAuthenticationMethod.Read
UserAuthenticationMethod.ReadWrite
UserAuthenticationMethod.Read.All
UserAuthenticationMethod.ReadWrite.All

Comment (mmcla) [reply concepts/permissions-reference.md]: Fixed.

**Comment by mmcla** [comment]
@dkershaw10 @Jackson-Woods @Lauragra I removed the do not merge tag mid-last-week; when should we expect this to merge?

**Comment by Lauragra** [comment]
@mmcla , please set the *ready to merge* label on PRs that you are ready to publish live, so they show up in my publishing query. I just applied the label to this PR and will process it today. Thanks.

**Comment by mmcla** [comment]
> @mmcla , please set the _ready to merge_ label on PRs that you are ready to publish live, so they show up in my publishing query. I just applied the label to this PR and will process it today. Thanks.

@Lauragra looks like Jackson has a comment I still need to address; working on that now.

**Comment by mmcla** [comment]
@Lauragra - Jackson and I are meeting at noon to review and we'll get back to you shortly after.

**Comment by mmcla** [comment]
> @Lauragra - Jackson and I are meeting at noon to review and we'll get back to you shortly after.

@Lauragra - All good, we locked changes.

**Pull request merged**